src/engine/uct/node/mod.rs

```diff
@@ -146,21 +148,40 @@ impl Node {
                 .iter()
                 .map(|&m| Node::new(m, self.config))
                 .collect();
+            if self.children.len() <= (game.size() * game.size() / 10) as usize {
+                let player = game.last_move().color().opposite();
+                if !self.config.uct.playout_aftermath || game.winner() == player {
+                    //don't pass if we're losing on the board on CGOS, but otherwise it's OK
+                    self.children.push(Node::new(Pass(player), self.config));
+                }
+            }
+            
             self.descendants = self.children.len();
         }
  }
 
     pub fn expand(&mut self, board: &Board) -> bool {
         let not_terminal = !board.is_game_over();
-        let mut children = vec![];
         if not_terminal && self.plays >= self.config.uct.expand_after {
-            children = board.legal_moves_without_eyes()
+            let mut children = board.legal_moves_without_eyes()
                 .iter()
                 .map(|m| self.new_leaf(board, m))
                 .collect();
+                
+            self.priors(&mut children, board);
+            self.children = children;
+            
+            if self.children.len() <= (board.size() * board.size() / 10) as usize {
+                let player = board.next_player();
+                if board.winner() == player {
```

Comment:
If you want to be able to turn off adding the pass move you will need to add the config check here as `expand_root()` won't be called if we reuse the subtree from the previous genmove.